src/handlers.rs

```diff
@@ -4,33 +4,261 @@ extern crate libc;
 extern crate url;
 
 use libc::c_char;
+use libc::uint32_t;
 use std::ffi::CStr;
+use std::ffi::CString;
+use std::fs::File;
+use std::io::Read;
 use url::Url;
 
-fn string_from_ptr(ptr: *const c_char) -> String {
-    let cstr = unsafe { CStr::from_ptr(ptr as *const i8) };
-    String::from(cstr.to_str().unwrap())
+// TODO(ry) Share this with the def in src/main.rs.
+#[repr(C)]
+pub struct DenoC {
+    _unused: [u8; 0],
+}
+
+// Also defined in src/assets.h
+#[repr(C)]
+pub struct deno_asset_t {
+    data: *const c_char,
+    len: uint32_t,
 }
 
+// TODO(ry) Share this extern block with those in main.rs.
+// See src/reply.h
+extern "C" {
+    pub fn deno_get_asset(filename: *const c_char) -> *const deno_asset_t;
+
+    pub fn reply_error(d: *const DenoC, cmd_id: uint32_t, msg: *const c_char);
+    pub fn reply_null(d: *const DenoC, cmd_id: uint32_t);
+    pub fn reply_code_fetch(
+        d: *const DenoC,
+        cmd_id: uint32_t,
+        module_name: *const c_char,
+        filename: *const c_char,
+        source_code: *const c_char,
+        output_code: *const c_char,
+    );
+}
+
+static SRC_DIR: &str = "/Users/rld/.deno/src/";
```

Comment:
Should SRC_DIR be using an env variable or something?

Reply:
Thanks - I added a TODO. It's a placeholder for future functionality - unused currently.